### DISCUSSION TITLE
How to stop getting logged out every time I migrate:fresh

### DISCUSSION THREAD

**Discussion by ghost** [filamentphp/filament#13674] [Help]
### Package

Panel builder

### Package Version

3

### How can we help you?

Every time I run `migrate:fresh` while developing, I am logged out of the session

So I have to re-login 100 times

Is there a way to keep me logged in even after running `migrate:fresh`?

I've tried removing the `AuthenticateSession` middleware from `AdminPanelProvider`, which did not help

**Comment by gggiii** [comment]
You can put this snippet in `routes/web.php` before initializing any routes:
```php
if (!app()->isProduction()) {
    $user = User::first();

    if ($user) {
        auth()->onceUsingId($user->id);
    }
}
```

**Reply by kjjd84** [reply]
Ended up putting this in my routes file which works perfectly:

```php
if (
    Request::is('admin/*') &&
    !App::isProduction() &&
    !Auth::check() &&
    $user = User::find(1)
) {
    Auth::login($user);
}
```

Thank you for the tip!

**Comment by dissto** [comment]
Alternatively, you could seed once normally like you did...then log in and tick the "remember me" checkbox. Then copy the remember token and the hashed password (not sure if that is needed) from your database for the user your want to login with and paste it into your user seeder.

So basically

```php
// default DatabaseSeeder.php
User::factory()->create([
    'name' => 'Test User',
    'email' => 'test@example.com',
    'password' => '$2y$12$EyJqKBgik5hHUhUQt28wOOuudbCfqS1EOxahdux6.NUuCcRfahtum', // password
    'remember_token' => 'xl7xodxuQA',
]);
```

You may need to change the session driver to `cookie` in your .env. Not sure if that also works with the database driver 🤔

```php
SESSION_DRIVER=cookie
```
